riscv-executor/src/lib.rs

```diff
@@ -433,7 +567,7 @@ mod builder {
 
         /// raw set next value of register by register index instead of name
         fn set_reg_idx(&mut self, idx: u16, value: Elem<F>) {
-            // Record register write in trace.
+            // Record register write in trace. Only for non-assignment registers.
```

Comment:
I guess this only happens for the pc now then?

Reply:
there are also the two `query_arg` registers

Reply:
ah yea